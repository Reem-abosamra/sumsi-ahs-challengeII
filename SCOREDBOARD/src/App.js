import { useState } from "react";
import styled from "styled-components";
import PlayerRow from "./components/PlayerRow/PlayerRow";
import RowContainer from "./components/RowContainer/RowContainer";

const App = () => {
  const [playerState, setPlayerState] = useState(["Hansi"]);
  const [currentInput, setCurrentInput] = useState("");

  return (
    <AppContainer>
      <h1>Scoreboard</h1>
      <ul>
        {playerState.map((player, index) => (
          <PlayerRow key={index} playerName={player} />
        ))}
      </ul>
      <RowContainer as="form">
        <div />
        <div>
          <StyledInput
            type="text"
            value={currentInput}
            placeholder="Enter your name"
            onChange={(event) => setCurrentInput(event.target.value)}
          />
        </div>
        <StyledButton
          onClick={(event) => {
            event.preventDefault();
            setPlayerState([...playerState, currentInput]);
            setCurrentInput("");
          }}
        >
          <span>Hinzuf√ºgen</span>
        </StyledButton>
      </RowContainer>
    </AppContainer>
  );
};

export default App;

const StyledButton = styled.button`
  cursor: pointer;
  user-select: none;
  transition: background-color 350ms ease;
  background-color: #0cca4a;
  border: none;
`;

const StyledInput = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 4px solid #dcdcdd;
  width: 100%;
  padding: 0.4rem 0px;
  font-size: 1.2rem;
  color: #dcdcdd;
  padding: 1rem;

  &::placeholder {
    color: #dcdcdd;
  }

  &:focus {
    outline-color: #dcdcdd;
  }
`;

const AppContainer = styled.main`
  border: 5px solid #1985a1;
  margin: 10rem auto;
  width: 50%;
  background-color: #46494c;
  border-radius: 1rem;
  padding: 2rem;
  text-align: center;
`;
