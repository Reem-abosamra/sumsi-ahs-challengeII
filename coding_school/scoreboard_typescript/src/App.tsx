import { FormEvent, FunctionComponent, useState } from "react"
import styled from "styled-components"
import PlayerRow from "./components/PlayerRow/PlayerRow"
import RowContainer from "./components/RowContainer/RowContainer"

const App: FunctionComponent = () => {
  const [playerState, setPlayerState] = useState<string[]>([])
  const [currentInput, setCurrentInput] = useState("")

  const deletePlayer = (index: number) => {
    const newPlayerState = playerState.filter(
      (_, i) => i !== index,
    )
    setPlayerState(newPlayerState)
  }

  const handleAddPlayer = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    setPlayerState([...playerState, currentInput])
    setCurrentInput("")
  }

  return (
    <main className="my-40 mx-auto w-1/2 rounded-2xl border-4 border-[#1985a1] bg-[#46494c] p-8 text-center">
      <h1>Scoreboard</h1>
      <ul>
        {playerState.map((player, index) => (
          <PlayerRow
            key={index}
            playerName={player}
            index={index}
            deletePlayer={deletePlayer}
          />
        ))}
      </ul>
      <RowContainer
        as="form"
        data-testid="playerName-form"
        onSubmit={handleAddPlayer}
      >
        <div />
        <div>
          <StyledInput
            type="text"
            value={currentInput}
            placeholder="Enter your name"
            onChange={(event) =>
              setCurrentInput(event.target.value)
            }
            data-testid="playerName-input"
          />
        </div>
        <StyledButton type="submit">
          <span>Hinzuf√ºgen</span>
        </StyledButton>
      </RowContainer>
    </main>
  )
}

export default App

const StyledButton = styled.button`
  cursor: pointer;
  user-select: none;
  transition: background-color 350ms ease;
  background-color: #0cca4a;
  border: none;
`

const StyledInput = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 5px solid #dcdcdd;
  width: 100%;
  padding: 0.4rem 0px;
  font-size: 1.2rem;
  color: #dcdcdd;
  padding: 1rem;

  &::placeholder {
    color: #dcdcdd;
  }

  &:focus {
    outline-color: #dcdcdd;
  }
`

const AppContainer = styled.main`
  border: 4px solid #1985a1;
  margin: 10rem auto;
  width: 50%;
  background-color: #46494c;
  border-radius: 1rem;
  padding: 2rem;
  text-align: center;
`
