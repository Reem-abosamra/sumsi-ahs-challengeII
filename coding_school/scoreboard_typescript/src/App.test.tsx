import renderer from "react-test-renderer"
import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import App from "./App"
import { createRenderer } from "react-dom/test-utils"

test("renders no player on start", () => {
  render(<App />)
  const playerRow = screen.queryByTestId("player-row-0")
  expect(playerRow).not.toBeInTheDocument()
})

test("adds a player on form submit", () => {
  const playerName = "Hansi"
  render(<App />)
  const playerNameInput = screen.getByTestId("playerName-input")
  fireEvent.change(playerNameInput, {
    target: { value: playerName },
  })
  const playerNameForm = screen.getByTestId("playerName-form")
  fireEvent.submit(playerNameForm)
  const playerRow = screen.getByTestId("player-row-0")
  expect(playerRow).toBeInTheDocument()
  expect(playerRow).toHaveTextContent(playerName)
})

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
