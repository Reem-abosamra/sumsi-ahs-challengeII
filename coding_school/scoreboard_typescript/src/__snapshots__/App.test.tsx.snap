// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="my-40 mx-auto w-1/2 rounded-2xl border-4 border-[#1985a1] bg-[#46494c] p-8 text-center"
>
  <h1>
    Scoreboard
  </h1>
  <ul />
  <form
    className="sc-bczRLJ cnyWMw"
    data-testid="playerName-form"
    onSubmit={[Function]}
  >
    <div />
    <div>
      <input
        className="sc-jSMfEi cwlJDZ"
        data-testid="playerName-input"
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
        value=""
      />
    </div>
    <button
      className="sc-eCYdqJ kLIsHT"
      type="submit"
    >
      <span>
        Hinzuf√ºgen
      </span>
    </button>
  </form>
</main>
`;
